File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/editprofile/EditProfileContract.java
package org.mifos.mobilewallet.mifospay.editprofile;

import org.mifos.mobilewallet.mifospay.base.BasePresenter;
import org.mifos.mobilewallet.mifospay.base.BaseView;

/**
 * Created by ankur on 27/June/2018
 */

public interface EditProfileContract {

    interface EditProfilePresenter extends BasePresenter {

        void fetchUserDetails();

        void handleNecessaryDataSave();

        void updateInputById(int id, String content);

        void updateEmail(String email);

        void updateMobile(String fullNumber);

        void handleProfileImageChangeRequest();

        void handleProfileImageRemoved();

        void handleClickProfileImageRequest();

        void handleExitOnUnsavedChanges();

        void onDialogNegative();

        void onDialogPositive();
    }

    interface EditProfileView extends BaseView<EditProfilePresenter> {

        void showDefaultImageByUsername(String fullName);

        void showUsername(String username);

        void showEmail(String email);

        void showVpa(String vpa);

        void showMobileNumber(String mobileNumber);

        void removeProfileImage();

        void changeProfileImage();

        void clickProfileImage();

        void onUpdateEmailError(String message);

        void onUpdateMobileError(String message);

        void showToast(String message);

        void showFab();

        void hideFab();

        void hideKeyboard();

        void startProgressBar();

        void stopProgressBar();

        void showDiscardChangesDialog();

        void closeActivity();
    }
}

File: mifospay/src/main/java/org/mifospay/editprofile/EditProfileContract.kt
package org.mifospay.editprofile

import org.mifospay.base.BasePresenter
import org.mifospay.base.BaseView

/**
 * Created by ankur on 27/June/2018
 */
interface EditProfileContract {
    interface EditProfilePresenter : BasePresenter {
        fun fetchUserDetails()
        fun handleNecessaryDataSave()
        fun updateInputById(id: Int, content: String?)
        fun updateEmail(email: String?)
        fun updateMobile(fullNumber: String?)
        fun handleProfileImageChangeRequest()
        fun handleProfileImageRemoved()
        fun handleClickProfileImageRequest()
        fun handleExitOnUnsavedChanges()
        fun onDialogNegative()
        fun onDialogPositive()
    }

    interface EditProfileView : BaseView<EditProfilePresenter?> {
        fun showDefaultImageByUsername(fullName: String?)
        fun showUsername(username: String?)
        fun showEmail(email: String?)
        fun showVpa(vpa: String?)
        fun showMobileNumber(mobileNumber: String?)
        fun removeProfileImage()
        fun changeProfileImage()
        fun clickProfileImage()
        fun onUpdateEmailError(message: String?)
        fun onUpdateMobileError(message: String?)
        fun showToast(message: String?)
        fun showFab()
        fun hideFab()
        fun hideKeyboard()
        fun startProgressBar()
        fun stopProgressBar()
        fun showDiscardChangesDialog()
        fun closeActivity()
    }
}